cmake/modules/zephyr_default.cmake

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# This CMake module will load all Zephyr CMake modules in correct order for
+# default Zephyr build system.
+#
+# Outcome:
+# See individual CMake module descriptions
+
+include_guard(GLOBAL)
+
+# The code line below defines the real minimum supported CMake version.
+#
+# Unfortunately CMake requires the toplevel CMakeLists.txt file to define the
+# required version, not even invoking it from a CMake module is sufficient.
+# It is however permitted to have multiple invocations of cmake_minimum_required.
+cmake_minimum_required(VERSION 3.20.0)
+
+message(STATUS "Application: ${APPLICATION_SOURCE_DIR}")
+
+find_package(ZephyrBuildConfiguration
+  QUIET NO_POLICY_SCOPE
+  NAMES ZephyrBuild
+  PATHS ${ZEPHYR_BASE}/../*
+  NO_CMAKE_PATH
+  NO_CMAKE_ENVIRONMENT_PATH
+  NO_SYSTEM_ENVIRONMENT_PATH
+  NO_CMAKE_PACKAGE_REGISTRY
+  NO_CMAKE_SYSTEM_PATH
+  NO_CMAKE_SYSTEM_PACKAGE_REGISTRY
+)
+
+# Prepare user cache
+include(python)
+include(ccache)
+
+# Load Zephyr extensions
+include(extensions)
+include(git)
+include(version)  # depends on hex.cmake
+
+#
+# Find tools
+#
+
+include(west)
+
+# Load default root settings
+include(root)
+
+#
+# Find Zephyr modules.
+# Those may contain additional DTS, BOARD, SOC, ARCH ROOTs.
+# Also create the Kconfig binary dir for generated Kconf files.
+#
+include(zephyr_module)
+
+include(boards)
+include(shields)
+include(arch)
+include(build_configuration)
+
+include(user_cache)
+include(verify-toolchain)
+include(host-tools)
+
+# Include board specific device-tree flags before parsing.
+include(${BOARD_DIR}/pre_dt_board OPTIONAL)
```

Comment:
This looks like it is now a module include, where it was a file include before. Did you mean to do that?

Reply:
no I did not.
Thanks for spotting.

As an extra note, this is part of the code that is kept like this because I didn't want to change existing behavior with this PR.

But when the new structure is in place then I intent to followup with extra cleanup and rework to better utilize the flexibility of the new scheme. But doing so will most likely introduce some changes in more parts of the build system so left for independent work.

Reply:
fixed